src/main/java/com/google/sps/data/Contest.java

```diff
@@ -26,41 +27,53 @@
 public abstract class Contest {
   public abstract String getName();
 
-  public abstract HashMap<Long, Candidate> getCandidates();
+  public abstract HashSet<String> getCandidates();
```

Comment:
I think someone added the Guava dependency already (some extra externally available Google Java libraries).  Can we switch this to use ImmutableSet instead?  Basically it prevents other people from adding to the Set after its been created. 

https://effectivejava.googleplex.com/third-edition#page=101

Reply:
I did some research and I found that the way to do this is
`Set<String> unmodifiableSet = Collections.unmodifiableSet(set);`
where `set` is a `HashSet<String>`.
How would this work with the AutoValue abstract methods, though?

Reply:
An example:
---
@AutoValue
public abstract class Animal {
  public abstract String name();
  public abstract int numberOfLegs();
  public abstract ImmutableSet<String> countries();

  public static Builder builder() {
    return new AutoValue_Animal.Builder();
  }

  @AutoValue.Builder
  public abstract static class Builder {
    public abstract Builder setName(String value);
    public abstract Builder setNumberOfLegs(int value);
    public abstract Builder setCountries(Set<String> value);
    public abstract Builder setCountries(String... value);
    public abstract Animal build();
  }
}

Reply:
Yikes that did not format well. Let me see if I can find a link.

Reply:
I think this site has some examples with autovalue with immutableset:
https://www.codota.com/code/java/classes/com.google.auto.value.AutoValue

But the idea is that you store immutableset, and you return immutableset in the actual AutoValue. You take in a regular set in the Builder.